src/assets/tabs.js

```diff
@@ -1,52 +1,92 @@
+   * through the "dehydrated" tab containers on the page and
+   * "hydrates" their tab link bar and contents sections, selecting
+   * the first tab by default.
+   * A click handler is registered for each tab link, which calls
+   * activateTab(). activateTab() is also run at the end of this
+   * function if there is a hash fragment in the URL, directing that
+   * the specified tab should be activated and brought into focus.
+   */
+
+  var containers = document.querySelectorAll("div.tabs-container");
+  containers.forEach(function(container) {
+    var ul = document.createElement("ul");
+    ul.classList.add(`${tabClass}-list`);
+    var sections = container.querySelectorAll("section");
+    sections.forEach(function(section, idx) {
+      var li = document.createElement("li");
+      li.classList.add(tabClass);
+      li.setAttribute("role", "tab");
+      li.setAttribute("aria-controls", section.id);
+      // Activate the first tab by default
+      if (idx == 0) {
+        section.style.display = "block";
+        li.classList.add(`${tabClass}--selected`);
+        li.setAttribute("aria-selected", "true");
+      } else {
+        section.style.display = "none";
+        li.setAttribute("aria-selected", "false");
+      }
+      li.appendChild(document.createTextNode(section.title));
+      li.addEventListener("click", function() {
+        activateTab(null, section.id);
       });
-      document.querySelector(
-        `section#${li.getAttribute("data-tab")}`
-      ).style.display =
-        "block";
+      ul.appendChild(li);
     });
+    container.querySelector("div.wrapper").children[0].appendChild(ul);
   });
-  // Show contents of first tab by default
-  lis[0].click();
+
+  if (window.location.hash) activateTab();
 }
 
-if (document.querySelector("#tabs")) {
-  // in case the document is already rendered
-  if (document.readyState != "loading") attachTabs();
-  // modern browsers
-  else if (document.addEventListener)
-    document.addEventListener("DOMContentLoaded", attachTabs);
-  // IE <= 8
-  else
-    document.attachEvent("onreadystatechange", function() {
-      if (document.readyState == "complete") attachTabs();
+function activateTab(hashEvent, tabID = null) {
+  /* This function is called when a tab link is clicked in any of
+   * the tab containers on the page, and also when the window's hash
+   * fragment changes. It activates the tab IDed in the parameter or the URL
+   * and, if invoked due to a hash fragment change, scrolls the window to the
+   * top of the proper tab container. The manual scrolling is ncessary because
+   * the placement of the tab IDs (in the section body, rather than the link)
+   * otherwise will cause the browser to scroll too far.
+   */
+
+  if (tabID == null && window.location.hash)
+    tabID = window.location.hash.slice(1);
+
+  var containers = document.querySelectorAll("div.tabs-container");
+  containers.forEach(function(container) {
+    var matchedSectionIndex = -1;
```

Comment:
If you were looking to optimize for efficiency, you could do all of this without much less looping, e.g.

```
  var section = document.getElementById(tabID);
  section.parentElement.children.forEach(
    function(_section) { _section.style.display = "none"; }
  );
  section.style.display = "block";
```

(Would be even more elegant with arrow functions, but it's probably best not to use them here, as this script is not transpiled,)

Reply:
Some of the looping is left over from when I was trying to do all of this without any element IDs... I'll update with this suggestion.

Reply:
Huh, `getElementById` seems to fail when the section in question has `display = "none"`. This, combined with the lack of convenient linkages between the tab headers and their content sections (requiring they be matched by their array indices), leads to most of the looping. It's not ideal, but it seems to work well enough, and a major re-implementation might be more work than it's worth, so leaving this for now.

Reply:
Agree with leaving it as it is (I've already merged), but surely `getElementById` can't be failing like that...  There must be something else going on.  Multiple (non-unique) IDs, perhaps?